### ISSUE TITLE
Vehicle Endpoints (e. g. vehicle_data) over the BLE-Interface?

### ISSUE THREAD

**Issue by haexpo** [teslamotors/vehicle-command#52]
Is it possible to get access to some of the "Vehicle Endpoints" (e. g. "vehicle_data") over the BLE-Interface?

**Comment by top-gun** [comment]
This is a must for many usecases.

I use the API to charge when I have PV surplus or when the grid power is cheap. But to control this task, I need to know the battery SOC and whether the cable is plugged. It would be bad to send commands otherwise, as it would keep the car awake and waste battery energy,

**Comment by haexpo** [comment]
Total blind flight without "vehicle_data"!

**Comment by top-gun** [comment]
> Total blind flight without "vehicle_data"!

Exactly. Atm, I get vehicle data in Home Assistant via the Tesla integration, but that's based on the old API and non-commercial. Having vehicle data via BLE - even if it's only basic stuff like SOC, charge port and wake/sleep - would allow me to completely migrate to the BLE side of the new API.

**Comment by BogdanDIA** [comment]
Hi, I have the same issue. How can I implement a proper automation with BLE commands if no status can be read from the car?

Any news on this ?

**Comment by i1skn** [comment]
I also think, that adding "vehicle_data" is a huge must have. Without it it's impossible to create BLE only applications, as you still need vehicle data to be loaded from the web.

**Comment by erikgieseler** [comment]
Is there any news on this?

**Comment by Maintraxx** [comment]
+1

**Comment by horvatbenjamin** [comment]
Any news? It would be great!

**Comment by m-chandler** [comment]
+1 this would awesome. Also using this for solar charging automation (as the Fleet API is a pain, and rate limited). Ideally I'd like to do this all with BLE.

At the moment I'm just manually turning off my solar charging automation when the car is plugged in / overnight, otherwise I'd be sending endless commands (keeping the car awake) as it's impossible to know the vehicle state via BLE.

**Comment by thefireblade** [comment]
Support was added in https://github.com/teslamotors/vehicle-command/pull/330, closing this issue
